### PR TITLE
#47 Secure settings

### PR DESCRIPTION
**Pull request by piotrgolasz**

### CODE REVIEW

**Comment by neo22s** [comment]
can someone test this please? I am unavailable. thanks

**Comment by piotrgolasz** [comment]
@svenbw Can you check the changes, or if any unittests have to be written?

**Comment by svenbw** [comment]
.htaccess and Cookie look okay to me.

But for Auth_Bcrypt I have some second thoughts: I think the code is functional (although I didn't test it) but it relies on the ORM module. Now the ORM module relies on Auth, that's the other way around. Maybe the ORM Auth part is a better place to support Bcrypt.
And secondly there are some syntactical issues (pure cosmetic): Check the code guidelines (spaces around the not, uppercase AND/OR).

**Comment by piotrgolasz** [comment]
@svenbw please re-test, thank you!

File: modules/orm/classes/Kohana/Auth/Bcrypt.php [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+ * @package    Kohana/Auth
+ * @author     Kohana Team
+ * @copyright  (c) 2007-2012 Kohana Team
+ * @license    http://kohanaframework.org/license
+ */
+class Kohana_Auth_Bcrypt extends Auth {
+
+    public function __construct($config = array())
+    {
+        if (!isset($config['cost']) or ! is_numeric($config['cost']) or $config['cost'] < 10)
```
Comment (svenbw) [inline modules/orm/classes/Kohana/Auth/Bcrypt.php]: Good checking, however some extra spaces needed between ! and uppercase OR's

**Comment by piotrgolasz** [comment]
@svenbw Requested changes amended, I also changed the hashing algorithm used in auth_autologin cookie generation - sha1 is considered broken.

**Pull request merged**